lib/modules/solidity/solcP.js

```diff
@@ -1,41 +1,58 @@
-let solc;
-
 const fs = require('fs-extra');
 const path = require('path');
 const constants = require('../../constants');
 const Utils = require('../../utils/utils');
 
-function findImports(filename) {
-  if (filename.startsWith('http') || filename.startsWith('git')) {
-    const fileObj = Utils.getExternalContractUrl(filename);
-    filename = fileObj.filePath;
-  }
-  if (fs.existsSync(filename)) {
-    return {contents: fs.readFileSync(filename).toString()};
+const ProcessWrapper = require('../../process/processWrapper');
+
+class SolcProcess extends ProcessWrapper {
+
+  findImports(filename) {
+    if (filename.startsWith('http') || filename.startsWith('git')) {
+      const fileObj = Utils.getExternalContractUrl(filename);
+      filename = fileObj.filePath;
+    }
+    if (fs.existsSync(filename)) {
+      return {contents: fs.readFileSync(filename).toString()};
+    }
+    if (fs.existsSync(path.join('./node_modules/', filename))) {
+      return {contents: fs.readFileSync(path.join('./node_modules/', filename)).toString()};
+    }
+    if (fs.existsSync(path.join(constants.httpContractsDirectory, filename))) {
+      return {contents: fs.readFileSync(path.join('./.embark/contracts', filename)).toString()};
+    }
+    return {error: 'File not found'};
   }
-  if (fs.existsSync(path.join('./node_modules/', filename))) {
-    return {contents: fs.readFileSync(path.join('./node_modules/', filename)).toString()};
+
+  loadCompiler(solcLocation) {
+    this.solc = require(solcLocation);
   }
-  if (fs.existsSync(path.join(constants.httpContractsDirectory, filename))) {
-    return {contents: fs.readFileSync(path.join('./.embark/contracts', filename)).toString()};
+
+  compile(jsonObj, cb) {
```

Comment:
Comments need to say that this is using the JSON standard io and probably contain a link to it somewhere.